client/selfrepair/normandy_driver.js

```diff
@@ -2,27 +2,44 @@ import uuid from 'node-uuid';
 
 import Mozilla from './uitour.js';
 
+export const STORAGE_DURABILITY_KEY = '_storageDurability';
+
 /**
  * Storage class that uses window.localStorage as it's backing store.
  */
-class LocalStorage {
+export class LocalStorage {
   /**
    * @param    {string} prefix Prefix to append to all incoming keys.
    */
   constructor(prefix) {
     this.prefix = prefix;
   }
 
+  async isDurable() {
+    const durabilityStatus = localStorage.getItem(STORAGE_DURABILITY_KEY);
+    return (parseInt(durabilityStatus, 10) >= 2);
+  }
+
   _makeKey(key) {
     return `${this.prefix}-${key}`;
   }
 
   async getItem(key) {
-    return localStorage.getItem(this._makeKey(key));
+    const storageIsDurable = await this.isDurable();
+    if (!storageIsDurable && !this.testing) {
```

Comment:
I changed this to `this.skipDurability`, and made the driver pass it in based on the value of `NormandyDriver::testing`.